pkg/solana/chainwriter/lookups.go

```
@@ -0,0 +1,354 @@
+
+	// Read derived lookup tables
+	for _, derivedLookup := range lookupTables.DerivedLookupTables {
+		// Load the lookup table - note: This could be multiple tables if the lookup is a PDALookups that resovles to more
+		// than one address
+		lookupTableMap, _, err := s.LoadTable(ctx, args, derivedLookup, s.reader, derivedTableMap)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error loading derived lookup table: %w", err)
+		}
+
+		// Merge the loaded table map into the result
+		for tableName, innerMap := range lookupTableMap {
+			if derivedTableMap[tableName] == nil {
+				derivedTableMap[tableName] = make(map[string][]*solana.AccountMeta)
+			}
+			for accountKey, metas := range innerMap {
+				derivedTableMap[tableName][accountKey] = metas
+			}
+		}
+	}
+
+	// Read static lookup tables
+	for _, staticTable := range lookupTables.StaticLookupTables {
+		addressses, err := getLookupTableAddresses(ctx, s.reader, staticTable)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error fetching static lookup table address: %w", err)
+		}
+		staticTableMap[staticTable] = addressses
+	}
+
+	return derivedTableMap, staticTableMap, nil
+}
+
+func (s *SolanaChainWriterService) LoadTable(ctx context.Context, args any, rlt DerivedLookupTable, reader client.Reader, derivedTableMap map[string]map[string][]*solana.AccountMeta) (map[string]map[string][]*solana.AccountMeta, []*solana.AccountMeta, error) {
+	// Resolve all addresses specified by the identifier
+	lookupTableAddresses, err := GetAddresses(ctx, args, []Lookup{rlt.Accounts}, nil, reader)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error resolving addresses for lookup table: %w", err)
+	}
+
+	// Nested map in case the lookup table resolves to multiple addresses
+	resultMap := make(map[string]map[string][]*solana.AccountMeta)
+	var lookupTableMetas []*solana.AccountMeta
+
+	// Iterate over each address of the lookup table
+	for _, addressMeta := range lookupTableAddresses {
```

Is it possible to get empty slice of lookupTableAddresses, if it is is that okay? If not maybe throw an error here

---

I think these checks should be sufficient, no? https://github.com/smartcontractkit/chainlink-solana/blob/9ea5350470c7f93bbcf2ff47af600f4c201faa89/pkg/solana/chainwriter/lookups.go#L346-L352